### DISCUSSION TITLE
Question: can you use this for runtime access of env vars for built apps?

### DISCUSSION THREAD

**Discussion by analyticd** [ElMassimo/vite-plugin-environment#10] [Q&A]
Thanks for this package. I am using it successfully to access env vars to a vite-built app which then gets served by nginx in a Docker environment. To comply with 12factorapp principles I want to be able to access the env vars at runtime not just at build time. Have you found an approach for that either using your package or another?

**Comment by ElMassimo** [comment]
Hi there!

Given that env variables are statically replaced in your built code, the only way to have "runtime" access to env variables, is to build in the server/environment that has the variables.

In your case, you would need to build in the same container that you plan to deploy.

Feel free to ask follow-up questions

Moving to discussions.

**Reply by analyticd** [reply]
Thanks so much for your reply. I really appreciate it. Yeah, that confirms what we are apparently finding. It is too bad because our devops (for good reason) likes to be able to deploy the same build to multiple environments while storing configuration elsewhere to be picked up by environment variables. If one tries to go the route of setting env vars on, say, the window object and including it in index.html, even then, though you can get the variables at runtime, browser caching can possibly make it so that you can't subsequently update them (if you wanted to) (ref: https://developers.redhat.com/blog/2021/03/04/making-environment-variables-accessible-in-front-end-containers# and comments). It is an interesting problem.

**Reply by munir131** [reply]
@analyticd We have similar issue, we want to use same container for multiple environment. We found this - https://github.com/micha149/vite-plugin-runtime-env
